include/xsimd/arch/generic/xsimd_generic_arithmetic.hpp

```diff
@@ -27,7 +27,7 @@ namespace xsimd
 
         // bitwise_lshift
         template <class A, class T, class /*=typename std::enable_if<std::is_integral<T>::value, void>::type*/>
-        inline batch<T, A> bitwise_lshift(batch<T, A> const& self, batch<T, A> const& other, requires_arch<generic>)
+        inline batch<T, A> bitwise_lshift(batch<T, A> const& self, batch<T, A> const& other, requires_arch<generic>) noexcept
         {
             return detail::apply([](T x, T y)
                                  { return x << y; },
```

Comment:
I think for MSVC to truly benefit here you'd need to also apply it to any lambdas too, e.g.
```cpp
return detail::apply([](T x, T y) noexcept
                                 { return x << y; },
```

(unlike `constexpr`, lambdas [aren't automatically `noexcept`](https://www.reddit.com/r/cpp_questions/comments/lusx7q/are_lambdas_automatically_noexcept/))

Reply:
Oh Joy :-)